File: app/src/main/java/com/omkar/jet2demo/data/local/AppDatabase.java
package com.omkar.jet2demo.data.local;

import io.realm.Realm;
import io.realm.RealmConfiguration;
import io.realm.RealmModel;


public class AppDatabase {

    private static AppDatabase dbHelper;
    private Realm realmInstance;

    private AppDatabase() {
        prepareRealmInstance();
    }

    /**
     * Gets instance.
     *
     * @return the instance
     */
    public static AppDatabase getInstance() {
        if (dbHelper == null) dbHelper = new AppDatabase();
        return dbHelper;
    }

    /**
     * Gets realm instance.
     *
     * @return the realm instance
     */
    private Realm getRealmInstance() {
        openDB();
        return realmInstance;
    }

    private void openDB() {
        if (realmInstance == null || realmInstance.isClosed())
            // Get mapView Realm instance for this thread with applied configuration
            realmInstance = Realm.getDefaultInstance();
    }

    private void prepareRealmInstance() {
        RealmConfiguration config = new RealmConfiguration.Builder().name("article.realm")
                .schemaVersion(1)
                .build();

        Realm.setDefaultConfiguration(config);
        Realm.compactRealm(config);

        openDB();
    }

    void closeDB() {
        if (realmInstance != null) realmInstance.close();
    }


    private void beginTransaction() {
        getRealmInstance().beginTransaction();
    }

    private void commitTransaction() {
        getRealmInstance().commitTransaction();
        Realm.compactRealm(Realm.getDefaultConfiguration());
    }

    public void saveArticleDataModel(RealmModel realmModel) {
        try {
            getRealmInstance().beginTransaction();
            getRealmInstance().copyToRealmOrUpdate(realmModel);
        } catch (Exception e) {
            e.printStackTrace();
        } finally {
            commitTransaction();
        }
    }


}

File: app/src/main/java/com/omkar/jet2demo/data/local/AppDatabase.kt
package com.omkar.jet2demo.data.local

import com.omkar.jet2demo.BuildConfig
import com.omkar.jet2demo.data.model.Image
import io.realm.Realm
import io.realm.RealmConfiguration
import io.realm.RealmList
import io.realm.RealmResults

class AppDatabase private constructor() {


    private fun prepareRealmInstance() {
        val config = RealmConfiguration.Builder().name("searchapp.realm")
            .schemaVersion(BuildConfig.SCHEMA_VERSION.toLong())
            .build()
        Realm.setDefaultConfiguration(config)
        Realm.compactRealm(config)
    }

    fun closeDB() {
        if (Realm.getDefaultInstance() != null) Realm.getDefaultInstance()!!.close()
    }


    fun getCommentForTheImage(imageId: String): String? {
        return Realm.getDefaultInstance()!!.where(Image::class.java)
            .equalTo("id", imageId).findFirst()?.comments
    }

    fun saveCommentForTheImage(imageId: String,comment: String) : Boolean{
        var image: Image? =
            Realm.getDefaultInstance()!!.where(Image::class.java)
                .equalTo("id", imageId).findFirst()
        if(image!=null){
            Realm.getDefaultInstance()!!.beginTransaction()
            image.comments = comment
            Realm.getDefaultInstance()!!.commitTransaction()
            return true
        }else{
            Realm.getDefaultInstance()!!.beginTransaction()
            image = Image()
            image.id = imageId
            image.comments = comment
            Realm.getDefaultInstance().copyToRealm(image)
            Realm.getDefaultInstance()!!.commitTransaction()
            return true
        }

        return false
    }

    companion object {
        private var appDatabase: AppDatabase? = null

        val instance: AppDatabase?
            get() {
                if (appDatabase == null)
                    appDatabase = AppDatabase()
                return appDatabase
            }
    }

    init {
        prepareRealmInstance()
    }
}